hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/BackgroundPipelineCreator.java

```diff
@@ -91,16 +96,20 @@ void triggerPipelineCreation() {
     scheduler.schedule(this::createPipelines, 0, TimeUnit.MILLISECONDS);
   }
 
-  private boolean skipCreation(HddsProtos.ReplicationFactor factor,
-                               HddsProtos.ReplicationType type,
-                               boolean autoCreate) {
-    if (type == HddsProtos.ReplicationType.RATIS) {
-      return factor == HddsProtos.ReplicationFactor.ONE && (!autoCreate);
-    } else {
+  private boolean skipCreation(ReplicationConfig replicationConfig,
+      boolean autoCreate) {
+    if (replicationConfig.getReplicationType()
+        == HddsProtos.ReplicationType.RATIS) {
+      return RatisReplicationConfig
+          .hasFactor(replicationConfig, ReplicationFactor.ONE) && (!autoCreate);
+    } else if (replicationConfig.getReplicationType()
+        == ReplicationType.STAND_ALONE) {
       // For STAND_ALONE Replication Type, Replication Factor 3 should not be
       // used.
-      return factor == HddsProtos.ReplicationFactor.THREE;
+      return ((StandaloneReplicationConfig) replicationConfig)
```

Comment:
factor != ONE make sense here?

Reply:
I tried to avoid modification in the algorithm, but I am fine with that. Just modified in d8d038af20c36cea571f7d6009dd9fbe1d78f579